server/eclipse-project/src/main/java/us/freeandfair/corla/crypto/PseudoRandomNumberGenerator.java

```diff
@@ -126,22 +126,22 @@ public PseudoRandomNumberGenerator(final String the_seed,
   /**
    * Generate the specified list of random numbers.
    * 
-   * @param the_from the "index" of the first random number generator to give
+   * @param the_from the "index" of the first random number to give
    * @param the_to the "index" of the final random number to give
    * 
-   * @return A list containing the_from - the_to + 1 random numbers
+   * @return A list containing the_to - the_from + 1 random numbers
    */
   //@ requires the_from <= the_to;
   // @todo kiniry Refine this specification to include public model fields.
-  // requires my_with_replacement;
-  // requires the_to <= my_maximum_index;
+  // requires my_with_replacement || the_to <= my_maximum_index;
```

Comment:
Why is this disjunction?

Reply:
If we're sampling with replacement, we don't need to worry about how many numbers we're asking for; if we're not, we do.